testing/src/test/java/org/oppia/android/testing/espresso/TextInputActionTest.kt

```diff
@@ -0,0 +1,62 @@
+package org.oppia.android.testing.espresso
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.espresso.TextInputAction.Companion.ErrorTextExisted
+import org.oppia.android.testing.espresso.TextInputAction.Companion.ErrorTextNotExisted
+import org.robolectric.annotation.LooperMode
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class TextInputActionTest {
+
+  @Test
+  fun testTextInputAction_hasErrorText_correctDescription() {
+    val matcher = TextInputAction.hasErrorText("Incorrect Administrator PIN. Please try again.")
+    assertThat(matcher.toString()).isEqualTo(
+      "The expected error text is 'Incorrect Administrator PIN. Please try again.'"
+    )
+  }
+
+  @Test
+  fun testTextInputAction_hasNoErrorText_correctDescription() {
+    val matcher = TextInputAction.hasNoErrorText()
+    assertThat(matcher.toString()).isEqualTo("There is no error text")
+  }
+
+  @Test
+  fun testTextInputAction_correctErrorText() {
+    val expectedErrorText = "This name is already in use by another profile."
+    val boundedMatcher = ErrorTextExisted(expectedErrorText)
+
+    val matcher = TextInputAction.hasErrorText("This name is already in use by another profile.")
+    assertThat(matcher.toString()).isEqualTo(boundedMatcher.toString())
+  }
+
+  @Test
+  fun testTextInputAction_incorrectErrorText() {
+    val expectedErrorText = "This name is already in use by another profile."
+    val boundedMatcher = ErrorTextExisted(expectedErrorText)
+
+    val matcher = TextInputAction.hasErrorText("Your PIN should be 5 digits long.")
+    assertThat(matcher.toString()).isNotEqualTo(boundedMatcher.toString())
+  }
+
+  @Test
+  fun testTextInputAction_noErrorText() {
+    val boundedMatcher = ErrorTextNotExisted()
+
+    val matcher = TextInputAction.hasNoErrorText()
+    assertThat(matcher.toString()).isEqualTo(boundedMatcher.toString())
+  }
+
+  @Test
+  fun testTextInputAction_noErrorText_errorTextExisted() {
+    val boundedMatcher = ErrorTextNotExisted()
+
+    val matcher = ErrorTextExisted("Your PIN should be 5 digits long.")
+    assertThat(matcher.toString()).isNotEqualTo(boundedMatcher.toString())
```

Comment:
Hmm. I still don't think this is quite right. Comparing the string result of the matcher doesn't actually tell us if the matcher works correctly.  We really need to verify that matchesSafely() & describeTo() are correct. The best way to do this would be to set up an test activity & layout and use it like in an actual Espresso environment, but I think it's fine just to call the two methods directly.

I suggest revamping all of these tests to call the relevant matcher methods directly. I also suggest marking both matcher classes as private in TextInputAction since they're technically implementation details, not part of the objec'ts public API.

Reply:
![Screenshot from 2021-09-24 03-19-37](https://user-images.githubusercontent.com/76056229/134635457-7f6dba2b-a8b2-473c-b8b1-3c44ae9b40c2.png)
@BenHenning. PTAL, if I am going in the right direction. Also the constructor for TextInputLayout is expecting type context. How do I pass context to it?

Reply:
@ayush0402 that's definitely on the right track. There are some small things that can be cleaned up there, but let's get it working first & I can follow up with suggestions once you have updated tests that are passing.

Since TextInputLayout is a view, you need to pass a context to it. For simplicity, you should be able to just use ``ApplicationProvider.getApplicationContext<Context>()`` to retrieve the context.

Reply:
The latest changes are much clearer, so resolving this.